### PR TITLE
feat: s2i builder quarkus support

### PR DESCRIPTION
**Pull request by lkingland**
- :gift: S2I Quarkus builder integration

The openjdk-17 S2I-enabled builder image does indeed appear to build and run Quarkus Functions when provided with a few magic environment variables.

/kind enhancement

Additional work towards #908

### CODE REVIEW

File: s2i/builder.go [hunk 1]
```diff
@@ -82,6 +83,19 @@ func (b *Builder) Build(ctx context.Context, f fn.Function) (err error) {
...
+	// S2I Builder image to trigger a build correctly.  This seems inconvenient
+	// to have here, but allows us to rely entirely on the shared S2I builder
+	// without needing to roll our own.
+	// Source: https://quarkus.pro/guides/deploying-to-openshift-s2i.html
+	if f.Runtime == "quarkus" {
+		cfg.Environment = append(cfg.Environment, api.EnvironmentSpec{
+			Name: "MAVEN_S2I_ARTIFACT_DIRS", Value: "target"})
+		cfg.Environment = append(cfg.Environment, api.EnvironmentSpec{
+			Name: "S2I_SOURCE_DEPLOYMENTS_FILTER", Value: "*-runner.jar lib"})
+	}
```
Comment (lance) [inline s2i/builder.go:97]: Can these not just exist as `buildEnvs` in func.yaml similar to https://github.com/knative-sandbox/kn-plugin-func/blob/main/templates/quarkus/manifest.yaml#L3-L9?

**Comment by matejvasek** [comment]
/hold for others
apart from Lance's comment /lgtm

File: templates/quarkus/manifest.yaml [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
   - name: BP_NATIVE_IMAGE
     value: "false"
   - name: BP_MAVEN_BUILT_ARTIFACT
     value: target/quarkus-app/lib/ target/quarkus-app/*.jar target/quarkus-app/app/ target/quarkus-app/quarkus/
   - name: BP_MAVEN_BUILD_ARGUMENTS
     value: package -DskipTests=true -Dmaven.javadoc.skip=true -Dquarkus.package.type=fast-jar
+  - name: MAVEN_S2I_ARTIFACT_DIRS
+    value: target
+  - name: S2I_SOURCE_DEPLOYMENTS_FILTER
+    value: *-runner.jar lib
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply templates/quarkus/manifest.yaml:11]: It's needed regenerate the zz_* file too.

**Pull request merged**